core_lib/src/managers/clipboardmanager.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+
+Pencil2D - Traditional Animation Software
+Copyright (C) 2005-2007 Patrick Corrieri & Pascal Naidon
+Copyright (C) 2012-2020 Matthew Chiawen Chang
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+*/
+#include "clipboardmanager.h"
+
+#include <QClipboard>
+
+#include <editor.h>
+
+ClipboardManager::ClipboardManager(Editor* editor) : BaseManager(editor, "ClipboardManager")
+{
+    resetStates();
+}
+
+ClipboardManager::~ClipboardManager()
+{
+
+}
+
+bool ClipboardManager::updateIfNeeded(const Layer* layer)
+{
```

Comment:
It was yes, because I needed the last stored position... however looking at the solution again, that was a bad idea. It's much cleaner to get the pos from outside. Fixed in latest commit.